src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_RefReadOnly.cs

```diff
@@ -570,25 +986,883 @@ public class ReadOnlyAttribute : System.Attribute { }
+            });
+        }
+
+        [Fact]
+        public void RefReadOnlyDefinitionsInsideUserDefinedIsReadOnlyAttribute_ClassInherit()
+        {
+            var code = @"
+namespace System.Runtime.CompilerServices
+{
+    public class IsReadOnlyAttribute : System.Attribute
+    {
+    }
+}
+public class Child : System.Runtime.CompilerServices.IsReadOnlyAttribute
+{
+    private int value;
+
+    public ref readonly int Method(ref readonly int x) => ref value;
+
+    public static int operator +(ref readonly Child x, ref readonly Child y) => 0;
+
+    public ref readonly int Property => ref value;
+
+    public ref readonly int this[ref readonly int x] => ref value;
+}";
+
+            CompileAndVerify(code, verify: false, symbolValidator: module =>
+            {
+                var type = module.ContainingAssembly.GetTypeByMetadataName("Child");
+
+                var method = type.GetMethod("Method");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, method.GetReturnTypeAttributes(), module.ContainingAssembly.Name);
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, method.Parameters.Single().GetAttributes(), module.ContainingAssembly.Name);
+
+                var @operator = type.GetMethod("op_Addition");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, @operator.Parameters[0].GetAttributes(), module.ContainingAssembly.Name);
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, @operator.Parameters[1].GetAttributes(), module.ContainingAssembly.Name);
+
+                var property = type.GetProperty("Property");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, property.GetAttributes(), module.ContainingAssembly.Name);
+
+                var indexer = type.GetProperty("this[]");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, indexer.GetAttributes(), module.ContainingAssembly.Name);
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, indexer.Parameters.Single().GetAttributes(), module.ContainingAssembly.Name);
+            });
+        }
+
+        [Fact]
+        public void RefReadOnlyDefinitionsInsideUserDefinedIsReadOnlyAttribute_ClassOverride()
+        {
+            var code = @"
+namespace System.Runtime.CompilerServices
+{
+    public abstract class IsReadOnlyAttribute : System.Attribute
+    {
+        public IsReadOnlyAttribute() { }
+
+        public abstract ref readonly int Method(ref readonly int x);
+
+        public abstract ref readonly int Property { get; }
+
+        public abstract ref readonly int this[ref readonly int x] { get; }
+    }
+}
+public class Child : System.Runtime.CompilerServices.IsReadOnlyAttribute
+{
+    private int value;
+
+    public override ref readonly int Method(ref readonly int x) => ref value;
```

Comment:
> public override ref readonly int Method(ref readonly int x) => ref value; [](start = 3, length = 74)

The override scenarios should be tested inside IsReadOnlyAttribute type. This test could be valuable too, but it doesn't test most interesting scenario.